src/Bicep.Core/TypeSystem/TupleType.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Bicep.Core.TypeSystem;
+
+/// <summary>
+/// Represents an array with a fixed number of items, each of which has an independently defined type.
+/// </summary>
+public class TupleType : ArrayType
+{
+    private readonly Lazy<ITypeReference> lazyItem;
+
+    public TupleType(string name, ImmutableArray<ITypeReference> items, TypeSymbolValidationFlags validationFlags) : base(name)
+    {
+        Items = items;
+        ValidationFlags = validationFlags;
+        // Items may contain DeferredTypeReferences, so only calculate Item on request
+        lazyItem = new(() => TypeHelper.CreateTypeUnion(this.Items));
```

Comment:
Use `LazyThreadSafetyMode.PublicationOnly` here?

Reply:
Also - we have caching in our declared & assigned type managers - are you sure that caching is also necessary here?

Reply:
The manager will be caching the `TupleType` instance, but without caching inside the type, `.Item` will be recalculated each time it is accessed.